core/domain/feedback_services.py

```diff
@@ -510,66 +510,69 @@ def get_thread_summaries(user_id, full_thread_ids):
     thread_summaries = []
     number_of_unread_threads = 0
     for index, thread in enumerate(threads):
-        feedback_thread_user_model_exists = (
-            feedback_thread_user_models[index] is not None)
-        if feedback_thread_user_model_exists:
-            last_message_read = (
-                last_two_messages[index][0].message_id
-                in feedback_thread_user_models[index].message_ids_read_by_user)
-        else:
-            last_message_read = False
-
-        if last_two_messages[index][0].author_id is None:
-            author_last_message = None
-        else:
-            author_last_message = user_services.get_username(
-                last_two_messages[index][0].author_id)
-
-        second_last_message_read = None
-        author_second_last_message = None
-
-        does_second_message_exist = (last_two_messages[index][1] is not None)
-        if does_second_message_exist:
+        if thread.original_author_id == user_id:
```

Comment:
I don't think this is the right place to do the filtering. Instead, we should only pass in the feedback threads that need to be retrieved, not fetch a lot of threads and then ignore (possibly most of) them. If you need a separate model to keep track of which threads have been started by the learner, it's fine to create one and populate it via a one-off job.